### ISSUE TITLE
aria-haspopup depricated from global attributes

### ISSUE THREAD

**Issue by curtbellew** [dequelabs/axe-core#4531]
### Product

axe-core

### Product Version

4.9.1

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
Does the fact that aria-haspopup has been deprecated from global attributes mean it should be reported as a violation when used in any roles not listed -  [https://www.w3.org/TR/wai-aria-1.2/#aria-haspopup](url) ?
#### Actual
When aria-haspopup is used on a DIV no violation is thrown

#### How to Reproduce
I would think the following code would return a violation but it does not.
`<div aria-haspopup="true">class</div>`

#### Additional context
NA

**Comment by straker** [comment]
Thanks for the question. We have a [few tickets](https://github.com/dequelabs/axe-core/issues/4433) currently [open](https://github.com/dequelabs/axe-core/issues/3341) about [deprecated ARIA roles and attributes](https://github.com/dequelabs/axe-core/issues/3253). In general we don't want to disallow attributes or roles where they were previously allowed as that would cause code that was previously working to then report violations. Instead what we want to do is indicate that the role or attribute has been deprecated for it's use and move the issue to probably a Needs Review issue.
